pkg/activator/net/revision_backends.go

```diff
@@ -220,8 +225,19 @@ func (rw *revisionWatcher) sendUpdate(clusterIP string, dests sets.String) {
 
 // checkDests performs probing and potentially sends a dests update. It is
 // assumed this method is not called concurrently.
-func (rw *revisionWatcher) checkDests(dests sets.String) {
-	if len(dests) == 0 {
+func (rw *revisionWatcher) checkDests(dests destSets) {
+	// TODO(taragu) the readyDests and notReadyDests should have different
+	// priorities of probing, because the first list is guaranteed to be good
+	// pods when clusterIP is probe-able, the ones in the second list are possibly
+	// not.
+	combinedDests := sets.NewString()
+	if len(dests.readyDests) != 0 {
+		combinedDests.Insert(dests.readyDests.List()...)
+	}
+	if len(dests.notReadyDests) != 0 {
+		combinedDests.Insert(dests.notReadyDests.List()...)
+	}
+	if len(dests.readyDests) == 0 && len(dests.notReadyDests) == 0 {
```

Comment:
I do not think this logic is correct. I think just `|readyDests|==0` should be sufficient.

Reply:
Hmm, should we still probe the not ready dests if ready dests is empty?

Reply:
that's the only reason for this PR, I think :-)

Reply:
@vagababov right 😄 does that mean we should keep this line as `if len(dests.readyDests) == 0 && len(dests.notReadyDests) == 0 {`? because otherwise we won't probe the not ready dests if ready dests is empty

Reply:
No. Since cluster IP is unhealthy is there are no readyDests by definition, even if readyDests is not empty.